src/applications/disability-benefits/all-claims/utils/submit.js

```diff
@@ -362,7 +364,46 @@ export const addForm4142 = formData => {
   return clonedData;
 };
 
+export const delete0781FormData = formData => {
+  // Workaround to avoid eslint rule around mutating params/args
+  const data = formData;
+  delete data.events;
+  delete data.eventTypes;
+  delete data.workBehaviors;
+  delete data.otherBehaviors;
+  delete data.healthBehaviors;
+  delete data.workBehaviorChanges;
+  delete data.socialBehaviorChanges;
+  delete data.behaviorsDetails;
+  delete data.supportingEvidenceRecords;
+  delete data.supportingEvidenceReports;
+  delete data.supportingEvidenceUnlisted;
+  delete data.supportingEvidenceWitness;
+  delete data.supportingEvidenceOther;
+  delete data.supportingEvidenceNoneCheckbox;
+  delete data.optionIndicator;
+  delete data.additionalRemarks781;
+  delete data.treatmentNoneCheckbox;
+  delete data.additionalInformation;
+  data.vaTreatmentFacilities.forEach((facility, index, object) => {
+    if (facility.treatmentLocation0781Related) {
+      object.splice(index, 1);
+    }
+  });
+  if (data.vaTreatmentFacilities.length === 0) {
+    delete data.vaTreatmentFacilities;
+  }
+};
+
 export const addForm0781 = formData => {
+  // In case of opt-out, we want to delete any 0781-related data
+  if (
+    formData.mentalHealthWorkflowChoice ===
```

Comment:
`mentalHealthWorkflowChoice` is a `view` property and, I think, has to be referenced as such, i.e., `formData['view:mentalHealthWorkflowChoice']`.

In addition, confirm that it's only the `OPT_OUT_OF_FORM0781` choice that necessitates deleting all 0781 data and **not also** the `SUBMIT_PAPER_FORM` option. I thought, from the https://github.com/department-of-veterans-affairs/va.gov-team/issues/105412#issuecomment-2754980134, that any option other than `COMPLETE_ONLINE_FORM` required form data deletion.